gateways/server/gcp-pubsub/validate.go

```diff
@@ -65,8 +66,5 @@ func validate(eventSource *v1alpha1.PubSubEventSource) error {
 	if eventSource.Topic == "" {
 		return fmt.Errorf("must specify topic")
 	}
```

Comment:
You want to check if the EnableWorkloadIdentity is set to true, if so, then  its fine to not have credentialsFile else it is required to have the credentialsFile.

Reply:
fixed